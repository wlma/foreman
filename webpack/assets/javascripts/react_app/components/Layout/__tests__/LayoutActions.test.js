import { testActionSnapshotWithFixtures } from '../../../common/testHelpers';
import {
  initializeLayout,
  showLoading,
  hideLoading,
  collapseLayoutMenus,
  expandLayoutMenus,
} from '../LayoutActions';

const fixtures = {
  'should initialize the layout': () =>
    initializeLayout({
      items: 'some items',
      isCollapsed: false,
      organization: 'org1',
      location: 'loc2',
    }),

  'should showLoading': () => showLoading(),

  'should hideLoading': () => hideLoading(),

  'should expandLayoutMenus': () => expandLayoutMenus(),

  'should collapseLayoutMenus': () => collapseLayoutMenus(),
};

describe('Layout actions', () => testActionSnapshotWithFixtures(fixtures));
