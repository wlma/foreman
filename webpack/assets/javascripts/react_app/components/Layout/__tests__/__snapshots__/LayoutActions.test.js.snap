// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout actions should collapseLayoutMenus 1`] = `
Object {
  "type": "LAYOUT_COLLAPSE",
}
`;

exports[`Layout actions should expandLayoutMenus 1`] = `
Object {
  "type": "LAYOUT_EXPAND",
}
`;

exports[`Layout actions should hideLoading 1`] = `
Object {
  "type": "LAYOUT_HIDE_LOADING",
}
`;

exports[`Layout actions should initialize the layout 1`] = `
Object {
  "payload": Object {
    "isCollapsed": false,
    "items": "some items",
    "location": "loc2",
    "organization": "org1",
  },
  "type": "LAYOUT_INITIALIZE",
}
`;

exports[`Layout actions should showLoading 1`] = `
Object {
  "type": "LAYOUT_SHOW_LOADING",
}
`;
